crates/types/src/schema/metadata/updater/mod.rs

```diff
@@ -289,7 +290,8 @@ impl SchemaUpdater {
 
     pub fn add_deployment(
         &mut self,
-        deployment_metadata: DeploymentMetadata,
+        deployment_address: DeploymentAddress,
+        mut deployment_metadata: DeploymentMetadata,
```

Comment:
I know this `deployment_address` here is messy, because it mostly copies over what's inside this `DeploymentMetadata`, but i needed it to pass around the routing header. I'll fix this in subsequent PRs (my plan is to completely remove `DeploymentMetadata` and a bunch of others).